Realm/RLMUtil.mm

```diff
@@ -454,7 +476,15 @@ id RLMMixedToObjc(realm::Mixed const& mixed,
         case realm::type_UUID:
             return [[NSUUID alloc] initWithRealmUUID:mixed.get<realm::UUID>()];
         default:
-            @throw RLMException(@"Invalid data type for RLMPropertyTypeAny property.");
+            if (mixed.is_type(realm::type_Dictionary)) {
+                return [[RLMManagedDictionary alloc] initWithParent:obj property:property parentInfo:*classInfo];
+            }
+            else if (mixed.is_type(realm::type_List)) {
+                return [[RLMManagedArray alloc] initWithParent:obj property:property parentInfo:*classInfo];
+            }
```

Comment:
`type_Dictionary` and `type_List` are not included in realm::DataType, in consequence this is showing a very annoying warning if used as a case, and that's the reason we are using a condition instead of adding a case here. @jedelbo any reason for this?